quickwit-core/src/index.rs

```diff
@@ -99,11 +98,42 @@ pub async fn delete_index(
         .mark_splits_as_deleted(index_id, split_ids)
         .await?;
 
-    let files = garbage_collect(metastore.as_ref(), index_id, storage_resolver).await?;
+    let file_entries = garbage_remove(metastore.as_ref(), index_id, storage_resolver).await?;
     //TODO: discuss & fix possible data race
     metastore.delete_index(index_id).await?;
 
-    Ok(files)
+    Ok(file_entries)
+}
+
+/// Prepare all danglings files for removal from the index specified with `index_id`.
+/// It also performs the removal.
+///
+/// * `metastore_uri` - The metastore Uri for accessing the metastore.
+/// * `index_id` - The target index Id.
+///
+pub async fn garbage_collect_index(
+    metastore_uri: &str,
+    index_id: &str,
+) -> anyhow::Result<Vec<FileEntry>> {
+    let metastore = MetastoreUriResolver::default()
+        .resolve(&metastore_uri)
+        .await?;
+    let storage_resolver = StorageUriResolver::default();
+
+    // schedule all staged splits for delete
+    let staged_splits = metastore
+        .list_splits(index_id, SplitState::Staged, None)
```

Comment:
I wonder if we should garbage collect "staged + older than 1h", just to be sure we are not removing staged splits that are about to be uploaded...

Right now our metastore is not concurrent, but it will eventually be.

Reply:
Indeed you are right, I think we need to have a concept of `updated_at` timestamp on the split to check when was the status updated and ignore the new ones

Reply:
that sounds good yes. Can you open an issue and take over that?

Reply:
I was planning to handle it here, but yes separating always gives more context to the ticket. I will add test and clean do we can close this. I already added TODO